commands/onboard.go

```diff
@@ -0,0 +1,212 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/authority"
+	"github.com/smallstep/certificates/ca"
+	"github.com/smallstep/certificates/pki"
+	"github.com/smallstep/cli/command"
+	"github.com/smallstep/cli/crypto/randutil"
+	"github.com/smallstep/cli/errs"
+	"github.com/smallstep/cli/ui"
+	"github.com/smallstep/cli/utils"
+	"github.com/urfave/cli"
+)
+
+// defaultOnboardingURL is the production onboarding url, to use a development
+// url use:
+//   export STEP_CA_ONBOARDING_URL=http://localhost:3002/onboarding/
+const defaultOnboardingURL = "https://api.smallstep.com/onboarding/"
+
+type onboardingConfiguration struct {
+	Name     string `json:"name"`
+	DNS      string `json:"dns"`
+	Address  string `json:"address"`
+	password []byte
+}
+
+type onboardingPayload struct {
+	Fingerprint string `json:"fingerprint"`
+}
+
+type onboardingError struct {
+	StatusCode int    `json:"statusCode"`
+	Message    string `json:"message"`
+}
+
+func (e onboardingError) Error() string {
+	return e.Message
+}
+
+func init() {
+	command.Register(cli.Command{
+		Name:      "onboard",
+		Usage:     "configure and run step-ca from the onboarding guide",
+		UsageText: "**step-ca onboard** <token>",
+		Action:    onboardAction,
+		Description: `**step-ca onboard** configures step certificates using the onboarding guide.
+
+Open https://smallstep.com/onboarding in your browser and start the CA with the
```

Comment:
is it api.smallstep.com or smallstep.com? earlier you used api.smallstep.com

Reply:
https://smallstep.com/onboarding is the web app, api.smallstep.com is the endpoint step-ca uses to get the configuration.